test/test_utmpx.rs

```diff
@@ -0,0 +1,41 @@
+use nix::utmpx::*;
+
+/// Test opening and rewinding the DB.
+#[test]
+fn test_utmpx_open_rewind() {
+    let _m = ::UTMPX_MTX
+        .lock()
+        .expect("Mutex got poisoned by another test");
+    unsafe {
+        let mut db = Utmp::open().unwrap();
+        db.rewind();
+    }
+}
+
+/// Test iterating through default DB.
+#[test]
+#[cfg(all(
+    any(target_arch = "x86", target_arch = "x86_64"),
```

Comment:
You should add a comment explaining why this test is only enabled for certain architectures.